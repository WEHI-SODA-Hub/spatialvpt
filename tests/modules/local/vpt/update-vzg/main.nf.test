nextflow_process {

    name "Test Process UPDATE_VZG"
    script "modules/local/vpt/update-vzg/main.nf"
    process "UPDATE_VZG"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'small_set' ],
                    [ file('${projectDir}/tests/data/202305010900_U2OS_small_set_VMSC00000_region_0.vzg', checkIfExists: true) ],
                    [ file('${projectDir}/tests/data/cellpose_micron_space.parquet', checkIfExists: true) ],
                    [ file('${projectDir}/tests/data/cell_by_gene_repartitioned.csv', checkIfExists: true) ],
                    [ file('${projectDir}/tests/data/cell_metadata_resegmented.csv', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
