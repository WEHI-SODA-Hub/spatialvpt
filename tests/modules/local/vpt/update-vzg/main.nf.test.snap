{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "small_set"
                        },
                        "202305010900_U2OS_small_set_VMSC00000_region_0_resegmented.vzg:md5,2bc513a74dd57d0d304e912310ca44aa"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ecc22ceda10ffcf3b0ed6811af376e7d"
                ],
                "versions": [
                    "versions.yml:md5,ecc22ceda10ffcf3b0ed6811af376e7d"
                ],
                "vzg_file": [
                    [
                        {
                            "id": "small_set"
                        },
                        "202305010900_U2OS_small_set_VMSC00000_region_0_resegmented.vzg:md5,2bc513a74dd57d0d304e912310ca44aa"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-08-09T13:40:53.475653386"
    }
}